packages/sdk/schema/src/types/projection.ts

```diff
@@ -150,33 +135,21 @@ export class ViewProjection {
       invariant(property);
       invariant(format);
 
-      if (isRename) {
-        delete this._schema.jsonSchema.properties![sourcePropertyName!];
-      }
-
-      let $id;
-      let reference;
+      const jsonProperty: JsonSchemaType = {};
       if (referenceSchema) {
-        $id = JSON_SCHEMA_ECHO_REF_ID;
+        setSchemaReference(jsonProperty, referenceSchema);
```

Comment:
Using the like this nullifies my previous comment, but still makes this function a poor abstraction, perhaps it would be better to do: `Object.assign(jsonProperty, getSchemaReference(referenceSchema))`

Reply:
let's discuss i'm not sure about this.